cmd/umserver/graceful_shutdown.go

```diff
@@ -0,0 +1,49 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"io"
+	"log"
+	"net/http"
+	"sync"
+	"time"
+)
+
+// closingStructure includes all components, which are connected with our application
+type closingStructure struct {
+	components []io.Closer
+}
+
+func gracefulShutdown(timeout time.Duration, wg *sync.WaitGroup, srv *http.Server, closers *closingStructure) error {
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+
+	// Shutdown HTTP server
+	go func() {
+		for _, component := range closers.components {
+			err := component.Close()
+			if err != nil {
+				log.Println("component error: %w", err)
+			}
+		}
+
+		err := srv.Shutdown(ctx)
```

Comment:
To stop serving requests, propose to close server first

Reply:
Thank you. I corrected it